snippets/ArrayWrapper1D.hpp

```diff
@@ -46,14 +46,14 @@ class ArrayWrapper1D : public std::array<T, N>
   template <typename B = T, typename std::enable_if<std::is_same<B, char>::value, int>::type = 0>
   operator std::string() const
   {
-    return std::string( this->data(), N );
+    return std::string( this->data(), std::min( N, strlen( this->data() ) ) );
```

Comment:
This works, but the strlen call can take a variable amount of time depending if there is no null character in the buffer. It's also UB to read beyond the buffer, which may trip UBSAN for some. How about using something like strnlen?

Reply:
I suppose the right function to call would be strlen_s: https://en.cppreference.com/w/c/string/byte/strlen

Reply:
Actually, scratch using strlen_s, as it's [not widely available](https://stackoverflow.com/questions/66346502/which-is-most-standard-strnlen-or-strnlen-s). I would stick to strnlen, which seems to be available in all C++ compilers.

Reply:
You're right, `strnlen` was the function I was looking for.
Thanks for your review!